src/contract.rs

```diff
@@ -44,67 +59,98 @@ pub fn init<S: Storage, A: Api, Q: Querier>(
-    }
 
     let init_config = msg.config();
     let admin = msg.admin.unwrap_or(env.message.sender);
-    let canon_admin = deps.api.canonical_address(&admin)?;
 
-    let mut total_supply: u128 = 0;
-    {
-        let initial_balances = msg.initial_balances.unwrap_or_default();
-        for balance in initial_balances {
-            let balance_address = deps.api.canonical_address(&balance.address)?;
-            let amount = balance.amount.u128();
-            let mut balances = Balances::from_storage(&mut deps.storage);
-            balances.set_account_balance(&balance_address, amount);
-            if let Some(new_total_supply) = total_supply.checked_add(amount) {
-                total_supply = new_total_supply;
-            } else {
-                return Err(StdError::generic_err(
-                    "The sum of all initial balances exceeds the maximum possible total supply",
-                ));
-            }
-            store_mint(
-                &mut deps.storage,
-                &canon_admin,
-                &balance_address,
-                balance.amount,
-                msg.symbol.clone(),
-                Some("Initial Balance".to_string()),
-                &env.block,
-            )?;
-        }
-    }
+    let total_supply: u128 = 0;
 
     let prng_seed_hashed = sha_256(&msg.prng_seed.0);
 
+    // Set stake config
+    let staked_token_decimals: u8;
+    if let Some(decimals) = msg.decimals {
+        staked_token_decimals = decimals;
+    } else {
+        staked_token_decimals = token_info_query(
+            &deps.querier,
+            256,
+            msg.staked_token.code_hash.clone(),
+            msg.staked_token.address.clone(),
+        )?
+        .decimals;
+    }
+
     let mut config = Config::from_storage(&mut deps.storage);
     config.set_constants(&Constants {
         name: msg.name,
-        symbol: msg.symbol,
-        decimals: msg.decimals,
-        admin: admin.clone(),
+        symbol: "STKD-".to_string() + &msg.symbol,
+        decimals: staked_token_decimals,
+        admin,
         prng_seed: prng_seed_hashed.to_vec(),
         total_supply_is_public: init_config.public_total_supply(),
-        deposit_is_enabled: init_config.deposit_enabled(),
-        redeem_is_enabled: init_config.redeem_enabled(),
-        mint_is_enabled: init_config.mint_enabled(),
-        burn_is_enabled: init_config.burn_enabled(),
         contract_address: env.contract.address,
     })?;
     config.set_total_supply(total_supply);
     config.set_contract_status(ContractStatusLevel::NormalRun);
-    let minters = if init_config.mint_enabled() {
-        Vec::from([admin])
-    } else {
-        Vec::new()
-    };
-    config.set_minters(minters)?;
 
-    Ok(InitResponse::default())
+    // Set distributors
+    Distributors(msg.distributors.unwrap_or_default()).save(&mut deps.storage)?;
+    DistributorsEnabled(msg.limit_transfer).save(&mut deps.storage)?;
```

Comment:
I think these "distributors" is unnecessary, in a panic situation where we need to disable the contract we can just shut it down. Is there a reason we would want to allow specific users to do things with tokens that we are disallowing from everyone else? Seems this is just a "centralized" mode without a specific use case.

Reply:
Distributors could work as another way to create a migration contract, or if we want to add a way to receive / distribute staked tokens in the future

Reply:
The distributors arent needed to migrate tokens, unless we for some reason wanted to stop users from doing anything other than migration with their tokens which is not necessary

Reply:
Since staked SHD will be non-transferable this allows for a future fix if we ever need them to be transferable